### DISCUSSION TITLE
WS2812B works fine, then only uses 6 LED´s

### DISCUSSION THREAD

**Discussion by Muyagi** [awawa-dev/HyperHDR#1007] [Q&A] [answered]
Hi,

on my PC monitor i put 142 LED´s. With WLED they work fine, all the time. I changed over to HyperSerialESP32 because it´s a lot smoother. My problem acutal is, the strip works fine for an undefined time most, max a few minutes. Then the strip freezes and only uses first 6 LED´s. I need so switch off strip and ESP to get it to work again. 

HyperHDR V20 WinX64
ESP32 @921600 Baud
WS2812B 142 LED

Any advice? 

Thanks in advance,
Chris

**Comment by awawa-dev** [comment]
Hi
And if you power off & on the frozen led device in the remote tab what is the effect?

**Reply by Muyagi** [reply]
Thank you for the fast respond. 

Only the first 6 leds turn off. They work as expected.

**Reply by awawa-dev** [reply]
I would usually say that you need to add a level shifter, although the symptom is unusual. Usually the first LED amplifies the signal unless it is too noisy. It is possible, however, that you will have to start with this or LED no. 6/7 are damaged. Do you have a common ground between the esp32 and the LED strip?

**Reply by Muyagi** [reply]
I have a level shifter included. It’s the same board. I only swap the esp with one that has wled on it. All grounds are joined together. 
Maybe the esp has a fault…
Let me test that in a few days. I’ll let you know the result.

**Reply by awawa-dev** [reply, marked as answer]
I doubt it's esp32's fault, because it doesn't hang, i.e. there's a reaction from the first 6 LEDs. If it did hang, there would be no reaction, and the AWA protocol ensures that it received the entire data packet, not just for the first 6 LEDs. You can flash WLED to esp32, it also supports the classic Adalight, although without AWA. And the LED rendering method is different.

**Reply by Muyagi** [reply]
Thanks. I will test the different options and let you know what I can find out.

**Reply by awawa-dev** [reply]
What  level shifter you use?

**Reply by Muyagi** [reply]
SN74AHCT125N

I always tried to use your recommendations. ;)

**Reply by Muyagi** [reply]
It kept bugging me, so I tried some things instead of sleeping. 

Different ESP, no difference. Flashed HyperWLED, works like charm with 921600 Baud. I can live with that. Thanks for your help. 

PS: I left AWA on, it works. Should it be unchecked?
